File: src/commands/list_command.cpp
#include "commands/list_command.hpp"

#include <plog/Log.h>

ListCommand::ListCommand(const std::shared_ptr<StationsDao> stationsDao, 
                         const std::shared_ptr<SettingsDao> settingsDao,
                         const std::shared_ptr<MediaPlayer> mediaPlayer)
    : Command("list", "List all available stations", stationsDao, settingsDao, mediaPlayer) {
}

void ListCommand::execute(const std::vector<std::string>& args) {
    m_cli.parse(args);

    if (m_cli.hasOption('h')) {
        LOG(plog::info) << m_cli.usage();
        return;
    }

    m_stationsDao->open(m_cli.getValue('f', getDefaultFile()));
    const auto ids = m_stationsDao->getAllIds();

    for (const auto& id : ids) {
        const auto station = m_stationsDao->findById(id);
        if (station != nullptr) {
            LOG(plog::info) << "id:" << std::to_string(station->getId())
                << ", name:\"" << station->getName()
                << "\", genre:\"" << station->getGenre()
                << "\", country:\"" << station->getCountry() << "\"";
        } else {
            LOG(plog::warning) << "No station found for id:" << id;
        }
    }

    m_stationsDao->close();
}

File: cora/list_command.py
import logging

import command

logger = logging.getLogger(__name__)


class ListCommand(command.Command):

    def __init__(self, stations, settings, player, subparsers):
        super().__init__(stations, settings, player)
        parser = subparsers.add_parser('list', help='List all available stations')
        parser.set_defaults(func=self.execute)

    def execute(self, args):
        ids = self.stations.get_all_ids()
        for id in ids:
            station = self.stations.find_by_id(id[0])
            logger.info("id:%s, name:%s, genre:%s, country:%s", str(station.id),
                        station.name, station.genre, station.country)
